internal/cli/certificate.go

```diff
@@ -0,0 +1,403 @@
+	"syscall"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	apiclient "github.com/flightctl/flightctl/internal/api/client"
+	"github.com/flightctl/flightctl/internal/client"
+	fccrypto "github.com/flightctl/flightctl/internal/crypto"
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+)
+
+const secondsInDay = 86400
+
+type CertificateOptions struct {
+	GlobalOptions
+	Name          string
+	Expiration    string
+	OutputFormat  string
+	OutputDir     string
+	EncryptOutput bool
+}
+
+func DefaultCertificateOptions() *CertificateOptions {
+	return &CertificateOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		OutputFormat:  "reference",
+		EncryptOutput: false,
+	}
+}
+
+func NewCmdCertificate() *cobra.Command {
+	o := DefaultCertificateOptions()
+	cmd := &cobra.Command{
+		Use: "certificate [subcommand]",
+		// more subcommands may be added later
+		Short:     "request a new enrollment certificate for a device with 'certificate request'",
+		Args:      cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		ValidArgs: []string{"request"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *CertificateOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Name, "name", "n", "csr", "specify a name for the certificate signing request")
+	fs.StringVarP(&o.Expiration, "expiration", "x", "7d", "specify desired certificate expiration in days, example: 7d")
+	fs.StringVarP(&o.OutputFormat, "output-format", "o", "reference", "specify desired output format: either 'agent-config' to output values in one config file, or 'components' to output cert, key, and ca values in separate files")
+	fs.StringVarP(&o.OutputDir, "output-dir", "d", "bin/agent/etc/flightctl/certs/", "specify desired output directory for key, cert, and ca files")
+	fs.BoolVarP(&o.EncryptOutput, "encrypt", "e", false, "option to encrypt key file with a password from env var $FCPASS, or if $FCPASS is not set password must be provided during runtime")
+}
+
+func (o *CertificateOptions) Complete(cmd *cobra.Command, args []string) error {
+	return o.GlobalOptions.Complete(cmd, args)
+}
+
+func (o *CertificateOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	if o.OutputFormat != "reference" && o.OutputFormat != "embedded" {
+		return fmt.Errorf("invalid output format specified: %s\noutput format must be either `reference` (agent config printed to stdout references key, cert, CA files written out during runtime) or `embedded` (agent config printed to stdout has key, cert, CA info embedded)", o.OutputFormat)
+	}
```

Comment:
This would mean users can no longer request certificates without agent-config or that are not for enrollment.
We can do that, but should then `certificat request` no longer seems appropriate as command and we should probably pivot back to `enrollmentconfig`.

Reply:
Hm, yes. Okay, I will make it so they can get an enrollment cert or some other cert, and in case of enrollment cert it has output format options of reference or embedded.

Reply:
Higher-level question: if this CLI command is going to be used for non-enrollment certs, those will not be auto-approved, so should this process block until the cert is ready? That doesn't seem to make sense. Maybe I'll include the blocking and checking for the cert only in the case that it is an enrollment cert request

Reply:
Actually, looking back at the title of this PR, I think the original goal as I had understood it was to offer a command to get enrollment certs only. This makes sense because these are auto-approved so the command can wait for the certificates to be populated. This wouldn't work with other types of certs unless we want to extend the auto-approval functionality to cover more types - which may not be a good idea.